### PR TITLE
fix: Fixed table_type for GOVERNED tables

### PR DESCRIPTION
**Pull request by svdimchenko**
# Description

with following table

```
{'Name': 'test', 'DatabaseName': 'raw', 'Owner': 'hadoop', 'CreateTime': datetime.datetime(2022, 3, 8, 15, 49, 45, tzinfo=tzlocal()), 'UpdateTime': datetime.datetime(2022, 3, 8, 15, 49, 45, tzinfo=tzlocal()), 'LastAccessTime': datetime.datetime(1970, 1, 1, 1, 0, tzinfo=tzlocal()), 'Retention': 0, 'StorageDescriptor': {'Columns': [{'Name': 'brand', 'Type': 'string'}, {'Name': 'dt', 'Type': 'timestamp'}], 'Location': 's3://stage-dwh-transformed/brand_governed', 'InputFormat': 'org.apache.hadoop.hive.ql.io.parquet.MapredParquetInputFormat', 'OutputFormat': 'org.apache.hadoop.hive.ql.io.parquet.MapredParquetOutputFormat', 'Compressed': False, 'NumberOfBuckets': -1, 'SerdeInfo': {'SerializationLibrary': 'org.apache.hadoop.hive.ql.io.parquet.serde.ParquetHiveSerDe', 'Parameters': {'serialization.format': '1'}}, 'BucketColumns': [], 'SortColumns': [], 'Parameters': {}, 'SkewedInfo': {'SkewedColumnNames': [], 'SkewedColumnValues': [], 'SkewedColumnValueLocationMaps': {}}, 'StoredAsSubDirectories': False}, 'PartitionKeys': [], 'TableType': 'GOVERNED', 'Parameters': {'transient_lastDdlTime': '1646750985', 'classification': 'parquet', 'table_type': 'LAKEFORMATION_GOVERNED'}, 'CreatedBy': 'xxx', 'IsRegisteredWithLakeFormation': False, 'CatalogId': 'xxx', 'IsMultiDialectView': False}
```

I've got an error on `dbt docs generate`:

```
Encountered an error while generating catalog: 'GOVERNED'
```

I've added GOVERNED table type and improved logging as well

## Checklist

- [x] You followed [contributing section](https://github.com/dbt-athena/dbt-athena#contributing)
- [x] You kept your Pull Request small and focused on a single feature or bug fix.
- [x] You added unit testing when necessary
- [x] You added functional testing when necessary

### CODE REVIEW

File: dbt/adapters/athena/impl.py [hunk 1]
```diff
@@ -583,7 +582,7 @@ def _get_one_table_for_non_glue_catalog(
                     "column_comment": col.get("Comment", ""),
                 },
             }
-            for idx, col in enumerate(table["Columns"] + table.get("PartitionKeys", []))
+            for idx, col in enumerate(table.get("Columns", []) + table.get("PartitionKeys", []))
```
Comment (nicor88) [inline dbt/adapters/athena/impl.py:585]: for the sake of knowledge, what are the scenarios where Columns is not there?

**Review by nicor88 — DISMISSED** [review]

File: dbt/adapters/athena/impl.py (hunk 1 above)
Comment (svdimchenko) [reply dbt/adapters/athena/impl.py:585]: TBH "Columns" key is almost absent for TableTypeDef class. I don't use external catalogs, so I'm not sure if it works at all )

**Comment by svdimchenko** [comment]
> @svdimchenko have a look at the CI, there are some issues that needs to be addressed.

I see there is problem with https://github.com/dbt-athena/dbt-athena/blob/49896c21361ea74bb28aa1ee184c19f01d746776/tests/unit/test_adapter.py#L417-L417

do we have a use case when we need to create table without type ?

**Comment by nicor88** [comment]
As far as I remember there are specific scenarios (like external catalogs) where you can have tables without type, so yes, we should cover that scenarios.

File: dbt/adapters/athena/impl.py (hunk 1 above)
Comment (svdimchenko) [reply dbt/adapters/athena/impl.py:585]: just that fact that there is no Columns key in mentioned class:
```
TableTypeDef = TypedDict(
    "TableTypeDef",
    {
        "Name": str,
        "DatabaseName": NotRequired[str],
        "Description": NotRequired[str],
        "Owner": NotRequired[str],
        "CreateTime": NotRequired[datetime],
        "UpdateTime": NotRequired[datetime],
        "LastAccessTime": NotRequired[datetime],
        "LastAnalyzedTime": NotRequired[datetime],
        "Retention": NotRequired[int],
        "StorageDescriptor": NotRequired[StorageDescriptorOutputTypeDef],
        "PartitionKeys": NotRequired[List[ColumnOutputTypeDef]],
        "ViewOriginalText": NotRequired[str],
        "ViewExpandedText": NotRequired[str],
        "TableType": NotRequired[str],
        "Parameters": NotRequired[Dict[str, str]],
        "CreatedBy": NotRequired[str],
        "IsRegisteredWithLakeFormation": NotRequired[bool],
        "TargetTable": NotRequired[TableIdentifierTypeDef],
        "CatalogId": NotRequired[str],
        "VersionId": NotRequired[str],
        "FederatedTable": NotRequired[FederatedTableTypeDef],
        "ViewDefinition": NotRequired[ViewDefinitionTypeDef],
        "IsMultiDialectView": NotRequired[bool],
    },
)
```

Comment (nicor88) [reply dbt/adapters/athena/impl.py:585]: check [this](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/glue/client/get_table.html), **Columns** is in `StorageDescriptor `, as we do in [989b6b80bb7a2ce0abb5ace7545a215a2f79/dbt/adapters/athena/impl.py#L563](https://github.com/dbt-athena/dbt-athena/blob/3c66989b6b80bb7a2ce0abb5ace7545a215a2f79/dbt/adapters/athena/impl.py#L563), I believe that we should use `table.get("StorageDescriptor", {}].get("Columns", [])`

Comment (svdimchenko) [reply dbt/adapters/athena/impl.py:585]: so I just don't want to get an error if I start using such external catalogs. And the error is possible in this case (just based on python code)

Comment (svdimchenko) [reply dbt/adapters/athena/impl.py:585]: > check [this](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/glue/client/get_table.html), **Columns** is in `StorageDescriptor `, as we do in [989b6b80bb7a2ce0abb5ace7545a215a2f79/dbt/adapters/athena/impl.py#L563](https://github.com/dbt-athena/dbt-athena/blob/3c66989b6b80bb7a2ce0abb5ace7545a215a2f79/dbt/adapters/athena/impl.py#L563), I believe that we should use `table.get("StorageDescriptor", {}].get("Columns", [])`

I thought about that as well, but it worked somehow before...I just didn't want to break it now

**Comment by nicor88** [comment]
@svdimchenko final comment, https://github.com/dbt-athena/dbt-athena/pull/661#discussion_r1617360774 the rest looks good, thanks for the refactor.

**Review by nicor88 — APPROVED** [review]

**Pull request merged**